lib/tests/streamlit/type_util_test.py

```diff
@@ -109,22 +119,116 @@ def test_data_frame_with_dtype_values_to_bytes(self):
+        ]
+    )
+    def test_is_colum_type_arrow_incompatible(
+        self, column: pd.Series, incompatible: bool
+    ):
+        self.assertEqual(
+            type_util.is_colum_type_arrow_incompatible(column),
+            incompatible,
+            f"Expected {column} to be {'incompatible' if incompatible else 'compatible'} with Arrow.",
+        )
+
+    @parameterized.expand(
+        [
+            (pd.Series([1, 2, "3"]), True),
+            # Complex numbers:
+            (pd.Series([1 + 2j, 3 + 4j, 5 + 6 * 1j]), True),
+            # Timedelta:
+            (pd.Series([pd.Timedelta("1 days"), pd.Timedelta("2 days")]), True),
+            # Decimal:
+            (pd.Series([Decimal("1.1"), Decimal("2.2")]), True),
+            # Mixed-integer types:
+            (pd.Series([1, 2, "3"]), True),
+            # Mixed:
+            (pd.Series([1, 2.1, "3", True]), True),
+            # timedelta64
+            (pd.Series([np.timedelta64(1, "D"), np.timedelta64(2, "D")]), True),
+            # Frozenset:
+            (pd.Series([frozenset([1, 2]), frozenset([3, 4])]), True),
+            # Dicts:
+            (pd.Series([{"a": 1}, {"b": 2}]), True),
+            # Complex types:
+            (pd.Series([TestObject(), TestObject()]), True),
+            # Supported types:
+            (pd.Series([1, 2, 3]), False),
+            (pd.Series([1, 2, 3.0]), False),
+            (pd.Series(["foo", "bar"]), False),
+            (pd.Series([True, False, None]), False),
+            (pd.Series(["foo", "bar", None]), False),
+            (pd.Series([[1, 2], [3, 4]]), False),
+            (pd.Series(["a", "b", "c", "a"], dtype="category"), False),
+            (pd.Series([date(2020, 1, 1), date(2020, 1, 2)]), False),
+        ]
+    )
+    def test_fix_arrow_incompatible_column_types(
+        self, column: pd.Series, incompatible: bool
+    ):
+        """Test that `fix_arrow_incompatible_column_types` correctly fixes
+        columns containing unsupported types by converting them to string and
+        leaves supported columns unchanged.
         """
+        df = pd.DataFrame({"c1": column})
+        fixed_df = fix_arrow_incompatible_column_types(df)
+        col_dtype = fixed_df["c1"].dtype
+        inferred_type = infer_dtype(fixed_df["c1"])
+
+        if incompatible:
+            # Column should have been converted to string.
+            self.assertEqual(col_dtype, "object")
+            self.assertEqual(inferred_type, "string")
+        else:
+            # Column should have the original type.
+            self.assertEqual(col_dtype, df["c1"].dtype)
+            self.assertEqual(inferred_type, infer_dtype(df["c1"]))
+
+    def test_fix_no_columns(self):
+        """Test that `fix_arrow_incompatible_column_types` does not
+        modify a DataFrame if all columns are compatible with Arrow.
+        """
+
         df = pd.DataFrame(
             {
-                "complex": [1 + 2j, 3 + 4j, 5 + 6 * 1j],
```

Comment:
was there a reason why this was removed?

Reply:
It's a different test than before: ` test_fix_no_columns` should not fix any columns and `complex` columns would be unsupported. Git is just detecting those changes a bit misleading.